apps/web/ui/links/destination-url-input.tsx

```diff
@@ -31,6 +46,31 @@ export const DestinationUrlInput = forwardRef<
   ) => {
     const inputId = useId();
     const { isMobile } = useMediaQuery();
+    const [debouncedUrl] = useDebounce(inputProps.value, 500);
+    const [duplicateLinks, setDuplicateLinks] = useState<string[]>([]);
+    const { id: workspaceId } = useWorkspace();
+    const [prevUrl, setPrevUrl] = useState<string | undefined>();
+    const isInitialMount = useRef(true);
+
+    useEffect(() => {
+      if (isInitialMount.current) {
+        isInitialMount.current = false;
+        return;
+      }
+
+      if (debouncedUrl && debouncedUrl !== prevUrl) {
+        setPrevUrl(debouncedUrl.toString());
+        fetch(
+          `/api/links/verify/destination-url?url=${encodeURIComponent(debouncedUrl.toString())}&workspaceId=${workspaceId}`,
+        )
+          .then((res) => res.json())
+          .then((data) => {
+            setDuplicateLinks(data.exists ? data.links : []);
+          });
```

Comment:
I think we need to add handling for **editing** a short link so that itself cannot be counted as a duplicate. We'll need to either update the `api/links/verify/destination-url` endpoint to accept a link ID to filter out, or move the `exists` determination to the client side and filter out any current link ID.